appserver/microprofile/health-glassfish/src/main/java/org/glassfish/microprofile/impl/HealthService.java

```diff
@@ -0,0 +1,31 @@
+package org.glassfish.microprofile.impl;
+
+
+import org.glassfish.api.StartupRunLevel;
+import org.glassfish.api.event.EventListener;
+import org.glassfish.hk2.runlevel.RunLevel;
+import org.glassfish.internal.api.Globals;
+import org.glassfish.internal.data.ApplicationInfo;
+import org.glassfish.internal.deployment.Deployment;
+import org.glassfish.microprofile.health.HealthReporter;
+import org.jvnet.hk2.annotations.Service;
+
+@Service(name = "healthcheck-service")
+@RunLevel(StartupRunLevel.VAL)
+public class HealthService implements EventListener {
```

Comment:
It's required to register the event listener, otherwise the event method won't be called:

```
import org.glassfish.api.event.Events;
import org.glassfish.hk2.api.PostConstruct;

public class HealthService implements PostConstruct, EventListener {

    @Inject
    Events events;

    @Override
    public void postConstruct() {
        events.register(this);
    }
```